### PR TITLE
[5.1] Delete schemaorg item after deleting an item

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #43830 .

### Summary of Changes
Added a onContentAfterDelete event to the schmea org Plugin 

### Testing Instructions
see issue #43830

### Actual result BEFORE applying this Pull Request
The schemaorg entry remains orphaned in the database

### Expected result AFTER applying this Pull Request
The schemaorg entry is deleted with the item

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/system/schemaorg/src/Extension/Schemaorg.php [hunk 1]
```diff
@@ -537,4 +538,30 @@ protected function isSupported($context)
 
         return false;
     }
+
+    /**
+     * The delete event.
+     *
+     * @param   Ojbject    $event  The event
```
Comment (sandewt) [inline plugins/system/schemaorg/src/Extension/Schemaorg.php]:
```suggestion
     * @param   Object    $event  The event
```
or 
`* @param   Model\AfterDeleteEvent  $event  The event`

File: plugins/system/schemaorg/src/Extension/Schemaorg.php [hunk 2]
```diff
@@ -537,4 +538,30 @@ protected function isSupported($context)
...
+        $itemId  = $event->getItem()->id;
+
+        $db    = $this->getDatabase();
+        $query = $db->getQuery(true);
+
+        $query->delete($db->quoteName('#__schemaorg'))
+            ->where($db->quoteName('itemId') . '= :itemId')
+            ->bind(':itemId', $itemId, ParameterType::INTEGER)
+            ->where($db->quoteName('context') . '= :context')
+            ->bind(':context', $context, ParameterType::STRING);
```
Comment (Quy) [inline plugins/system/schemaorg/src/Extension/Schemaorg.php]:
```suggestion
            ->where($db->quoteName('context') . '= :context')
            ->bind(':itemId', $itemId, ParameterType::INTEGER)
            ->bind(':context', $context, ParameterType::STRING);
```

Comment (chmst) [reply plugins/system/schemaorg/src/Extension/Schemaorg.php]: Thank you. To keep it DRY I've made an extra method with your change

**Comment by LadySolveig** [comment]
I have tested this item :white_check_mark: successfully on c0e319cb7d7ca0ad6aac720fa9ce6ede593fa60c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43839">issues.joomla.org/tracker/joomla-cms/43839</a>.</sub>

**Comment by bembelimen** [comment]
I have tested this item :white_check_mark: successfully on c0e319cb7d7ca0ad6aac720fa9ce6ede593fa60c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43839">issues.joomla.org/tracker/joomla-cms/43839</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**